### DISCUSSION TITLE
Question about "append instead of overwrite"

### DISCUSSION THREAD

**Discussion by Steamgjk** [Derecho-Project/derecho#233] [Q&A] [answered]
To continue the discussion:

I am playing with simple_replicated_objects_json
I notice that for the Foo part, every time a node ordered_send an update, it overwrites the value of Foo.

Here I want the value of Foo to be something like a log list/vector, e.g. every time a node ordered_send a request, this request is appended in the Foo vector, and derecho guarantees consistent log sequence across all nodes. How can I do this? Any pointers to the example code? Thanks.

Ken says
"Well, you just have to configure the group in persistent mode (not volatile mode). Now each new value is saved (logged) as a distinct new version. Later you can fetch the versions by version number, by specifying the version you wish to retrieve when you call get."

But here if I have low-performance disks and I want in-memory operation, is it still possible to do "appending"?
Or, is it possible to specify memory as the storage backend?

When you are saying "persistent mode", does it have to be really persistent storage? Can we just use memory as the storage backend?

**Comment by KenBirman** [comment, marked as answer]
Linux supports a feature called the /ramfs file system.  If you set this up and then run the system while in a folder located on /ramfs (copy the needed config files over), the versioning will occur yet everything will be held in memory.

**Reply by Steamgjk** [reply]
Hi, @KenBirman 
Is it called tmpfs or memfs?  https://www.kernel.org/doc/html/latest/filesystems/tmpfs.html

I googled it and it seems tmpfs is closer to what you are describing. memfs seems to be some projects in Node.js.

**Reply by KenBirman** [reply]
Sorry, it is actually called ramfs.  tmpfs is a normal disk storage, used for temporary files.  (answer updated)
